pallets/reward-distribution/src/tests.rs

```diff
@@ -33,3 +45,132 @@ fn test_set_rewards_per_block() {
+
+		assert_ok!(RewardDistribution::distribute_rewards(100, XCM(18)));
+	});
+}
+
+#[cfg(test)]
+#[test]
+fn on_initialize_hook_distribution_works() {
+	run_test(|| {
+		ext::pooled_rewards::get_total_stake_all_pools::<Test>.mock_safe(move || {
+			let initial_stakes = build_total_stakes();
+			MockResult::Return(Ok(initial_stakes))
+		});
+		ext::security::get_active_block::<Test>.mock_safe(move || MockResult::Return(1));
+
+		let mut expected_pool_ids = vec![XCM(0), XCM(1), XCM(2), XCM(3)].into_iter();
+		let mut expected_stake_per_pool = vec![46, 13, 34, 4].into_iter();
+		ext::pooled_rewards::distribute_reward::<Test>.mock_safe(
+			move |pool_id, reward_currency, amount| {
+				let expected_pool_id = expected_pool_ids.next().expect("More calls than expected");
+				let expected_stake_per_this_pool =
+					expected_stake_per_pool.next().expect("More calls than expected");
+				assert_eq!(pool_id, &expected_pool_id);
+				assert_eq!(amount, expected_stake_per_this_pool);
+				assert_eq!(
+					reward_currency,
+					&<Test as orml_currencies::Config>::GetNativeCurrencyId::get()
+				);
+				MockResult::Return(Ok(()))
+			},
+		);
+
+		System::set_block_number(100);
+		ext::security::get_active_block::<Test>.mock_safe(move || MockResult::Return(100));
+
+		let new_rewards_per_block = 100;
+		assert_ok!(RewardDistribution::set_reward_per_block(
+			RuntimeOrigin::root(),
+			new_rewards_per_block
+		));
+
+		assert_eq!(RewardDistribution::rewards_adapted_at(), Some(100));
+
+		System::set_block_number(101);
+		RewardDistribution::execute_on_init(101);
+		assert_eq!(RewardDistribution::native_liability(), Some(100));
+	})
+}
+
+#[cfg(test)]
+#[test]
+fn udpate_reward_does_not_trigger_incorrectly() {
+	run_test(|| {
+		ext::pooled_rewards::get_total_stake_all_pools::<Test>.mock_safe(move || {
+			let initial_stakes = build_total_stakes();
+			MockResult::Return(Ok(initial_stakes))
+		});
+
+		ext::security::parachain_block_expired::<Test>
+			.mock_safe(move |_, _| MockResult::Return(Ok(false)));
+
+		let new_rewards_per_block = 100;
+		assert_ok!(RewardDistribution::set_reward_per_block(
+			RuntimeOrigin::root(),
+			new_rewards_per_block
+		));
+
+		System::set_block_number(2);
+		RewardDistribution::execute_on_init(2);
+
+		//same reward per block is expected
+		assert_eq!(pallet::RewardPerBlock::<Test>::get(), Some(new_rewards_per_block));
+	})
+}
+
+#[cfg(test)]
+#[test]
+fn udpate_reward_per_block_works() {
+	run_test(|| {
+		ext::pooled_rewards::get_total_stake_all_pools::<Test>.mock_safe(move || {
+			let initial_stakes = build_total_stakes();
+			MockResult::Return(Ok(initial_stakes))
+		});
+		ext::security::parachain_block_expired::<Test>
+			.mock_safe(move |_, _| MockResult::Return(Ok(true)));
+
+		let new_rewards_per_block = 10000;
+		assert_ok!(RewardDistribution::set_reward_per_block(
+			RuntimeOrigin::root(),
+			new_rewards_per_block
+		));
+
+		System::set_block_number(200);
+		//Security::<Test>::set_active_block_number(200);
```

Comment:
It is confusing especially since the test was made, there has been changes where now the active block is used for `RewardsAdapted`. I will mock `ext::security::get_active_block`. 
Also, I believe set_block_number is also setting the active block? Otherwise this test is not working properly.